app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,46 @@
+class CategoriesController < ApplicationController
+  def index
+    @category = Category.all
+  end
+
+  def show
+    @category = Category.find(params[:id])
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+
+    @category.save
+    redirect_to categories_path
+  end
+
+  def edit
+    @category = Category.find(params[:id])
+  end
+
+  def update
+    @category = Category.find(params[:id])
+
+    if @category.update_attributes(category_params)
+      flash[:notice] = 'successfully updated!'
```

Comment:
Please internationalize all strings. You can use the translate helper in a controller like that:
`I18n.t('translation_key')`